stripe/src/main/java/com/stripe/android/CustomerSession.kt

```diff
@@ -0,0 +1,537 @@
+package com.stripe.android
+
+import android.app.Activity
+import android.content.Context
+import android.os.Handler
+import androidx.annotation.RestrictTo
+import androidx.annotation.VisibleForTesting
+import androidx.localbroadcastmanager.content.LocalBroadcastManager
+import com.stripe.android.PaymentConfiguration.Companion.getInstance
+import com.stripe.android.Stripe.Companion.appInfo
+import com.stripe.android.exception.StripeException
+import com.stripe.android.model.Customer
+import com.stripe.android.model.PaymentMethod
+import com.stripe.android.model.ShippingInformation
+import com.stripe.android.model.Source
+import com.stripe.android.model.Source.SourceType
+import java.lang.ref.WeakReference
+import java.util.Calendar
+import java.util.concurrent.LinkedBlockingQueue
+import java.util.concurrent.ThreadPoolExecutor
+import java.util.concurrent.TimeUnit
+
+/**
+ * Represents a logged-in session of a single Customer.
+ *
+ * See [Creating ephemeral keys](https://stripe.com/docs/mobile/android/standard#creating-ephemeral-keys)
+ */
+open class CustomerSession @VisibleForTesting internal constructor(
+    context: Context,
+    keyProvider: EphemeralKeyProvider,
+    private val proxyNowCalendar: Calendar? = null,
+    private val threadPoolExecutor: ThreadPoolExecutor,
+    stripeRepository: StripeRepository,
+    publishableKey: String,
+    stripeAccountId: String?,
+    shouldPrefetchEphemeralKey: Boolean
+) {
+    @JvmSynthetic
+    internal var customerCacheTime: Long = 0
+    @JvmSynthetic
+    internal var customer: Customer? = null
+
+    private val operationIdFactory = StripeOperationIdFactory()
+    private val productUsage = CustomerSessionProductUsage()
+    private val listeners: MutableMap<String, RetrievalListener?> = mutableMapOf()
+    private val ephemeralKeyManager: EphemeralKeyManager
+
+    init {
+        val keyManagerListener = CustomerSessionEphemeralKeyManagerListener(
+            CustomerSessionRunnableFactory(
+                stripeRepository,
+                createHandler(),
+                LocalBroadcastManager.getInstance(context),
+                publishableKey,
+                stripeAccountId,
+                productUsage
+            ),
+            threadPoolExecutor, listeners, productUsage)
```

Comment:
how is ktlint ok w/ this crime that just occurred :sob:

Reply:
🤣🤣🤣